keras_hub/src/models/gemma/gemma_attention.py

```diff
@@ -106,12 +108,18 @@ def _apply_rope(self, x, start_index):
         )
         return x
 
-    def _can_use_flash_attention(self):
-        if not has_flash_attention_support():
+    def _use_fused_attention_op(self):
+        if not fused_attention_op_available():
             return False
         if self.dropout > 0.0:
             return False
-        if self.logit_soft_cap is None:
+        if running_on_gpu() and not gpu_supports_fused_attention_op():
```

Comment:
maybe more readable at this point to just split the gpu/tpu branches. double check my rewrite though!

```python
if running_on_gpu():
    # GPU never supports softcap in the fused op.
    if self.logit_soft_cap is not None:
        return False
    return gpu_supports_fused_attention_op()
elif running_on_tpu():
    # TPU supports softcap with on keras >= 3.10.
    sig = inspect.signature(ops.dot_product_attention)
    return "attn_logits_soft_cap" in sig.parameters
else:
    return False
```